### PR TITLE
Eject lathe materials

### PR DESCRIPTION
**Pull request by chromiumboy**
## About the PR

This is a continuation of PR #16398, originally created by [PilgrimViis](https://github.com/PilgrimViis). It adds a new tab to the UI of most lathes and techfabs from which you can reclaim stacks of material that were inserted into these machines (the ore processor and sheet-meister 2000 have been excluded since they are used to process raw materials into sheets)

It also makes some minor adjustments to the base lathe UI (it now has a slightly larger default size to account for the new button to open the reclamation UI, materials which have been used up are removed from the materials list rather than just stating 0 cm^3 remaining, and recipe tooltips now state their required materials in 'cm^3' for better clarity)

This PR will close issue #10896

**Media**

[Demonstration of material reclamation from lathes](https://youtu.be/b-EqpJAvRqA)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: chromiumboy and PilgrimViis
- add: Lathes and fabricators can now eject unused material sheets.

### CODE REVIEW

File: Content.Client/Content.Client.csproj [hunk 1]
```diff
@@ -23,6 +23,11 @@
     <ProjectReference Include="..\RobustToolbox\Robust.Client\Robust.Client.csproj" />
     <ProjectReference Include="..\Content.Shared\Content.Shared.csproj" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (Partmedia) [inline Content.Client/Content.Client.csproj:26]: Do XAML's typically need an explicit dependency to compile correctly?

File: Content.Client/Lathe/UI/LatheMaterialsEjectionMenu.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+using Content.Shared.Materials;
+using Robust.Client.AutoGenerated;
+using Robust.Client.GameObjects;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using System.Linq;
+
+namespace Content.Client.Lathe.UI
```
Comment (Partmedia) [inline Content.Client/Lathe/UI/LatheMaterialsEjectionMenu.xaml.cs:9]: Current conventions prefer file-scoped namespaces: https://docs.spacestation14.io/en/getting-started/conventions#file-layout. Please switch to those.

**Review by Partmedia — CHANGES_REQUESTED** [review]
Thank you for your contribution. This will be a useful change. I still need to test this in-game, but in the meantime I've pointed out a few minor things, if you wouldn't mind addressing those.

File: Content.Client/Content.Client.csproj (hunk 1 above)
Comment (chromiumboy) [reply Content.Client/Content.Client.csproj:26]: That code was auto-generated, so I have no clue

File: Content.Client/Lathe/UI/LatheMaterialsEjectionMenu.xaml.cs (hunk 2 above)
Comment (chromiumboy) [reply Content.Client/Lathe/UI/LatheMaterialsEjectionMenu.xaml.cs:9]: Good catch, I missed that

**Comment by UbaserB** [comment]
wooooo finally, what everyone has been waiting for!

**Review by Partmedia — APPROVED** [review]

**Pull request merged**